src/TestEnvironment.Docker/DockerEnvironment.cs

```diff
@@ -52,6 +56,74 @@ public void Dispose()
             }
         }
 
+        private async Task BuildRequiredImages(CancellationToken token)
+        {
+            foreach (var container in Dependencies.OfType<FromDockerfileContainer>())
+            {
+                // In order to pass the context we have to create tar file and use it as an argument.
+                var tempFileName = Guid.NewGuid().ToString();
+                using (var fileStream = new FileStream(tempFileName, FileMode.CreateNew))
+                {
+                    var tarArchive = TarArchive.CreateOutputTarArchive(fileStream);
+                    var contextDirectory = container.Context.Equals(".") ? Directory.GetCurrentDirectory() : container.Context;
+
+                    tarArchive.RootPath = contextDirectory.Replace('\\', '/');
+                    if (tarArchive.RootPath.EndsWith("/"))
+                    {
+                        tarArchive.RootPath = tarArchive.RootPath.Remove(tarArchive.RootPath.Length - 1);
+                    }
+
+
+                    AddDirectoryFilesToTar(tarArchive, contextDirectory, true);
+
+                    tarArchive.Close();
+                }
+
+                // Now call docker api.
+                using (var tarContextStream = new FileStream(tempFileName, FileMode.Open))
+                {
+                    var imageStream = await _dockerClient.Images.BuildImageFromDockerfileAsync(tarContextStream, new ImageBuildParameters
```

Comment:
Also maybe it would be useful to wrap such important docker api calls in try-catch blocks for logging purposes?

Reply:
During the testing I saw all docker related issues in the output, but I agree - to make it transparent on all the environments it is useful